stable/insights-chart/templates/insights-metrics-deployment.yaml

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2021 Red Hat, Inc.
+# Copyright Contributors to the Open Cluster Management project
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "insights.fullname" . }}-insights-metrics
+  labels:
+    app: {{ template "insights.name" . }}
+    chart: {{ template "insights.chart" . }}
+    component: "insights-metrics"
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "insights.name" . }}
+      component: "insights-metrics"
+      release: {{ .Release.Name }}
+      heritage: {{ .Release.Service }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "insights.name" . }}
+        ocm-antiaffinity-selector: "insightsmetrics"
+        component: "insights-metrics"
+        release: {{ .Release.Name }}
+        chart: {{ template "insights.chart" . }}
+        heritage: {{ .Release.Service }}
+    spec:
+      serviceAccountName: insights-client
+      securityContext:
+          runAsNonRoot: true
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: beta.kubernetes.io/arch
+                operator: In
+                values:
+                {{- range .Values.global.arch }}
+                - {{ . }}
+                {{- end }}
+        {{- with .Values.metrics.affinity.podAntiAffinity }}
+        podAntiAffinity: 
+{{ toYaml . | indent 10 }}
+        {{- end }}
+      tolerations:
+        - key: dedicated
+          operator: Exists
+          effect: NoSchedule
+        - effect: NoSchedule 
+          key: node-role.kubernetes.io/infra 
+          operator: Exists
+      containers:
+      - name: metrics 
+        image: {{ .Values.global.imageOverrides.insights_metrics }}
+        imagePullPolicy: {{ .Values.global.pullPolicy }}
+        securityContext:
+          privileged: false
+          readOnlyRootFilesystem: true
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+        args:
+        - --port=8080
+        - --telemetry-port=8081
+        resources:
+{{ toYaml .Values.insights.metrics.resources | indent 10 }}
+        livenessProbe:
+          exec:
+            command:
+            - ls
```

Comment:
Is there an API endpoint we could use instead?

Reply:
Yes , I am enabling TLS .. next PR @jlpadilla